ldap/servers/slapd/log.c

```diff
@@ -3363,31 +3488,42 @@ log__fix_rotationinfof(char *pathname)
                 break;
             case ACCESSLOG:
                 loginfo.log_numof_access_logs++;
+
                 break;
             case AUDITLOG:
                 loginfo.log_numof_audit_logs++;
                 break;
             }
         } else if (0 == strncmp(log_type, dirent->name, strlen(log_type)) &&
-                   (p = strrchr(dirent->name, '.')) != NULL &&
-                   15 == strlen(++p) &&
-                   NULL != strchr(p, '-')) { /* e.g., errors.20051123-165135 */
+                   (p = strchr(dirent->name, '.')) != NULL &&
+                   NULL != strchr(p, '-')) /* e.g., errors.20051123-165135 */
+        {
             struct logfileinfo *logp;
             char *q;
             int ignoreit = 0;
 
-            for (q = p; q && *q; q++) {
-                if (*q != '-' && !isdigit(*q))
+            for (q = ++p; q && *q; q++) {
+                if (*q != '-' &&
```

Comment:
I do not think that we speak of the same thing here:
  My point is that you now accept z and g as valid character in the middle of error file name rather than checking that it is at the end -;) 
  The ++ is ok as you moved it from the now removed strlen test to here, I will just add the q-p test to check that the date lenght is still ok ..

Reply:
@progier389 - all changes made, please review...